master/buildbot/process/buildstep.py

```diff
@@ -167,7 +167,9 @@ def _catchup(self):
         def next(x):
             self._catchup()
             return x
-        self.step._start_unhandled_deferreds.append(d)
+
+        if self.step._start_unhandled_deferreds is not None:
```

Comment:
_start_unhandled_deferreds shall never be none if this code is called.

if it is none, then there is another problem in the step (related to the new style VS old style)

http://docs.buildbot.net/latest/manual/new-style-steps.html